src/android/aidl/birthday_service/src/lib.rs

```diff
@@ -26,4 +30,61 @@ impl IBirthdayService for BirthdayService {
     fn wishHappyBirthday(&self, name: &str, years: i32) -> binder::Result<String> {
         Ok(format!("Happy Birthday {name}, congratulations with the {years} years!"))
     }
+    // ANCHOR_END: IBirthdayService
+
+    fn wishWithInfo(&self, info: &BirthdayInfo) -> binder::Result<String> {
+        Ok(format!(
+            "Happy Birthday {}, congratulations with the {} years!",
+            info.name, info.years,
+        ))
+    }
+
+    fn wishWithProvider(
+        &self,
+        provider: &Strong<dyn IBirthdayInfoProvider>,
+    ) -> binder::Result<String> {
+        Ok(format!(
+            "Happy Birthday {}, congratulations with the {} years!",
+            provider.name()?,
+            provider.years()?,
+        ))
+    }
+
+    fn wishWithErasedProvider(
+        &self,
+        provider: &SpIBinder,
+    ) -> binder::Result<String> {
+        use binder::binder_impl::Proxy;
```

Comment:
Isn't this re-exported by `binder`? If it's not but it's needed, we should fix that upstream.

Reply:
It does not currently appear to be re-exported.

Reply:
Never mind, see the next comment.